test/skimNtuple2016_HHbtag.cpp

```diff
@@ -272,56 +272,60 @@ int main (int argc, char** argv)
-  float c2g_rew = atof(argv[21]);
-  cout << "** INFO: kl, kt reweight " << kl_rew << " " << kt_rew << " [kt < -990 || kl < -990 : no HH reweight]" << endl;
-  cout << "**       c2, cg, c2g reweight " << c2_rew << " " << cg_rew << " " << c2g_rew << " [if any is < -990: will do only a klambda / kt reweight if requested]" << endl;
-
-  string susyModel = argv[22];
+  // reweight file according to NLO differential reweighting procedure https://gitlab.cern.ch/hh/eft-benchmarks
+  string EFTbm = argv[17];
+  string order = argv[18];
+  string uncertantie = "\"\"";
+  if (argv[19] != string("0")) uncertantie = argv[19];
+  bool cms_fake = false;
+  string opt20 (argv[20]);
+  if (opt20 == "1") cms_fake = true;
+  float kl_rew = atof(argv[21]);
+  float kt_rew = atof(argv[22]);
+  float c2_rew = atof(argv[23]);
+  float cg_rew = atof(argv[24]);
+  float c2g_rew = atof(argv[25]);
+  cout << "** INFO: EFT reweighting asked for benchmark " << EFTbm << " at order " << order << endl;
+  if (c2_rew > -999.0) cout << "** INFO: EFT reweighting overridden with coplings kl=" << kl_rew << " ; kt=" << kt_rew << " ; c2=" << c2_rew << " ; cg=" << cg_rew << " ; c2g=" << c2g_rew << " at order " << order << "[all -999 means no override; only c2!=-999 means only c2 overridden]" << endl;
+
+  string susyModel = argv[26];
   cout << "** INFO: requesting SUSY model to be: -" << susyModel << "- [NOTSUSY: no request on this parameter]" << endl;
 
   // external weight file for PUreweight - sample per sample
-  TString PUreweightFile = argv[23];
+  TString PUreweightFile = argv[27];
   cout << "** INFO: PU reweight external file: " << PUreweightFile << endl;
   
-  int DY_nJets  = atoi(argv[24]);
-  int DY_nBJets = atoi(argv[25]);
+  int DY_nJets  = atoi(argv[28]);
+  int DY_nBJets = atoi(argv[29]);
   cout << "** INFO: nJets/nBjets for DY bin weights: " << DY_nJets << " / " << DY_nBJets << endl;
-  int isDYI = atoi(argv[26]);
+  int isDYI = atoi(argv[30]);
   bool isDY = (isDYI == 1) ? true : false;
 
   // this value is read just for "consistency", but is never actually used,
   // since in 2016 there is no ttHToTauTau sample, hence this stitching is not needed
   bool isttHToNonBB = false;
-  int isttHToNonBBI = atoi(argv[27]);
+  int isttHToNonBBI = atoi(argv[31]);
   if (isttHToNonBBI == 1) isttHToNonBB = true;
   cout << "** INFO: isttHToNonBB: " << isttHToNonBB << endl;
 
   bool isHHNLO = false;
-  int isHHNLOI = atoi(argv[28]);
+  int isHHNLOI = atoi(argv[32]);
   if (isHHNLOI == 1) isHHNLO = true;
   cout << "** INFO: isHHNLO: " << isHHNLO << endl;
 
   // ------------------  decide what to do for the reweight of HH samples
   enum HHrewTypeList {
-    kNone      = 0,
-    kFromHisto = 1,
-    kDynamic   = 2
+    kNone    = 0, //no reweighting
+    kDiffRew = 1, //differential reweight using EFT benchmarks
+    kC2scan  = 2, //differential reweight for c2 scan
+    kOverRew = 3  //differential reweight with manual override of the coupling's values
```

Comment:
I haven't read everything yet but I have a question: is the old reweighting still possible? Or it has been completely eliminated?

Reply:
The possibility of applying the old reweighing procedure is not implemented. Following offline discussion, the old class is kept in the following two files: [HHReweight5D_Morion17.h](https://github.com/LLRCMS/KLUBAnalysis/blob/83c3a1131f3a3dd1d8cf81dba9f447b8e5c3b9ef/interface/HHReweight5D_Morion17.h) and [HHReweight5D_Morion17.cc](https://github.com/LLRCMS/KLUBAnalysis/blob/83c3a1131f3a3dd1d8cf81dba9f447b8e5c3b9ef/src/HHReweight5D_Moriond17.cc)